pkg/api/usecase/wishcard/interactor.go

```diff
@@ -0,0 +1,172 @@
+package wishcard
+
+import (
+	"context"
+	"time"
+	tagEntity "wantum/pkg/domain/entity/tag"
+	wishCardEntity "wantum/pkg/domain/entity/wishcard"
+	"wantum/pkg/domain/repository"
+	"wantum/pkg/domain/service/place"
+	"wantum/pkg/domain/service/tag"
+	"wantum/pkg/domain/service/wishcard"
+	"wantum/pkg/domain/service/wishcardtag"
+	"wantum/pkg/werrors"
+)
+
+type Interactor interface {
+	CreateNewWishCard(ctx context.Context, userID int, activity, description, place string, date *time.Time, categoryID int, tags []string) (*wishCardEntity.Entity, error)
+	UpdateWishCard(ctx context.Context, wishCardID, userID int, activity, description, place string, date, doneAt *time.Time, categoryID int, tags []string) (*wishCardEntity.Entity, error)
+	DeleteWishCardByID(ctx context.Context, wishCardID int) error
+	GetByID(ctx context.Context, wishCardID int) (*wishCardEntity.Entity, error)
+	GetByCategoryID(ctx context.Context, categoryID int) (wishCardEntity.EntitySlice, error)
+}
+
+type interactor struct {
+	masterTxManager      repository.MasterTxManager
+	wishCardService      wishcard.Service
+	tagService           tag.Service
+	placeService         place.Service
+	wishCardsTagsService wishcardtag.Service
+}
+
+func New(masterTxManager repository.MasterTxManager, wishCardService wishcard.Service, tagService tag.Service, placeService place.Service, wishCardsTagsService wishcardtag.Service) Interactor {
+	return &interactor{
+		masterTxManager:      masterTxManager,
+		wishCardService:      wishCardService,
+		tagService:           tagService,
+		placeService:         placeService,
+		wishCardsTagsService: wishCardsTagsService,
+	}
+}
+
+func (i *interactor) CreateNewWishCard(ctx context.Context, userID int, activity, description, place string, date *time.Time, categoryID int, tags []string) (*wishCardEntity.Entity, error) {
+
+	var newWishCard *wishCardEntity.Entity
+	err := i.masterTxManager.Transaction(ctx, func(ctx context.Context, masterTx repository.MasterTx) error {
+		place, err := i.placeService.Create(ctx, masterTx, place)
+		if err != nil {
+			// TODO: placeがすでにあったら無限に増えてしまう
+			return werrors.Stack(err)
+		}
+		var tagIDs []int
```

Comment:
Capacity定義してあげた方がよさそう
```suggestion
		tagIDs := make([]int, 0, len(tags))
```

Reply:
ナイス気づき...！ありがとう！